backend/src/main/java/com/rota/api/StaffService.java

```diff
@@ -94,4 +94,20 @@ public void removeStaff(int id) {
       staffRepository.save(staff);
     });
   }
+
+  /**
+   * Updates a staff member in the database.
+   *
+   * @param id           Of the staff member being updated.
+   * @param updatedStaff Updated staff details.
+   * @return The updated {@link Staff} member.
+   */
+  public Staff updateStaff(int id, StaffDto updatedStaff) {
+    // Check to see if that staff member exists, throw exception if not.
+    staffRepository.findById(id).orElseThrow(StaffNotFoundException::new);
```

Comment:
What happens when this exception is thrown? Does it crash the server?

Reply:
StaffNotFoundException is a RuntimeException, so it can be handled in an exception handler and respond with a 404 and a nice message at some point.

Reply:
OK, for the moment it gives 500 - internal server error, which is not very helpful.
A possible solution would be to throw the error and have a try-catch block inside the controller, but I'm curious to see how your proposed solution works.

Reply:
I've updated - now throws a 400 with a nice error message.